src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -513,7 +513,7 @@ static Vector128<float> HalfAsWidenedUInt32ToSingle_Vector128(Vector128<uint> va
                 bitValueInProcess = Vector128.ShiftLeft(bitValueInProcess, 13);
 
                 // Double the offsetMaskedExponentLowerBound if value is either Infinity or NaN
-                offsetMaskedExponentLowerBound = Vector128.ConditionalSelect(Vector128.Equals(infinityOrNaNMask, Vector128<uint>.Zero),
+                offsetMaskedExponentLowerBound = ElementWiseSelect(Vector128.Equals(infinityOrNaNMask, Vector128<uint>.Zero),
```

Comment:
I'd notably expect the JIT is already optimizing this case correctly.

It has heuristics to recognize cases where the `condition` is definitively a mask, and so for cases like this where the mask is passed in directly it can generally do the right thing.

The issue primarily comes about when the mask comes indirectly, such as via a cached/reused local or when the mask is done multiple times and results in the JIT producing a CSE, in which case it isn't smart enough (at least today) to do the relevant tracking that the input is `AllBitsSet` or `Zero` on a per-element basis.

Shouldn't hurt anything to be explicit, just figured it was worth a basic explanation of how it works today.